doc/modules/ensemble.rst

```diff
@@ -1095,6 +1095,44 @@ supported for multiclass context.
 
   * :ref:`sphx_glr_auto_examples_ensemble_plot_monotonic_constraints.py`
 
+.. _categorical_support_gbdt:
+
+Categorical Support
+-------------------
+
+For datasets with categorical data, :class:`HistGradientBoostingClassifier`
+and :class:`HistGradientBoostingRegressor` has native support for splitting
+on categorical features. This is often better than one hot encoding because
+it leads to faster training times and trees with less depth. When splitting
+a node, the categorical feature will be split into two subsets: one going to
+the left child and the other going to the right child. First, the histogram
+for each categorical feature is first sorted according to the ratio:
+`sum of gradient / sum of hessian` in each bin. Then the best split is found
+by considering splits along the stored histogram.
```

Comment:
I'd suggest to expand a little bit:

> The canonical way of considering categorical splits is to consider all of the :math`2^{K - 1} - 1` partitions where `K` is the number of categories. This can quickly become prohibitive when `K` is large.  Fortunately, since gradient boosting trees are always regression trees (even for classification problems), there exist a faster strategy that can yield equivalent splits. First, the categories of a feature are sorted according to the ratio `sum_gradient_k / sum_hessians_k` of each category `k`. Once the categories are sorted, one can consider *continuous partitions*, i.e. treat the categories as if they were ordered continuous values (see <Fisher ref here> for more details). As a result, only `k - 1` splits need to be considered instead of :math`2^{K - 1} - 1`.